internal/status/all_replicas_ready_test.go

```diff
@@ -153,32 +154,32 @@ var _ = Describe("AllReplicasReady", func() {
 			When("transitions to true", func() {
 				BeforeEach(func() {
 					sts.Status.ReadyReplicas = 5
-					sts.Status.Replicas = 5
+					sts.Spec.Replicas = &desiredReplicas
 				})
 
 				It("updates the transition time", func() {
-					condition := rabbitmqstatus.AllReplicasReadyCondition([]runtime.Object{sts}, existingCondition)
+					condition := rabbitmqstatus.AllReplicasReadyCondition([]runtime.Object{sts}, oldCondition)
 
-					Expect(existingCondition).NotTo(BeNil())
-					existingConditionTime := existingCondition.LastTransitionTime.DeepCopy()
+					Expect(oldCondition).NotTo(BeNil())
+					oldConditionTime := oldCondition.LastTransitionTime.DeepCopy()
 
-					Expect(condition.LastTransitionTime.Equal(existingConditionTime)).To(BeFalse())
-					Expect(condition.LastTransitionTime.Before(existingConditionTime)).To(BeFalse())
+					Expect(condition.LastTransitionTime.Equal(oldConditionTime)).To(BeFalse())
```

Comment:
Can we replace the matcher to use Equal instead of BeFalse and BeTrue? Is there a reason for not doing that?